hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/table/HoodieTableSource.java

```diff
@@ -520,10 +520,9 @@ private MergeOnReadInputFormat mergeOnReadInputFormat(
         this.schema.getColumnNames().toArray(new String[0]),
         this.schema.getColumnDataTypes().toArray(new DataType[0]),
         this.requiredPos,
-        this.conf.getString(FlinkOptions.PARTITION_DEFAULT_NAME),
         this.limit == NO_LIMIT_CONSTANT ? Long.MAX_VALUE : this.limit, // ParquetInputFormat always uses the limit value
+        this.conf,
         getParquetConf(this.conf, this.hadoopConf),
-        this.conf.getBoolean(FlinkOptions.UTC_TIMEZONE),
         this.internalSchemaManager
```

Comment:
Can we avoid to pass the conf around, it's a preference, in personal I don't like the way to pass around huge confuguration objects.

Reply:
Ok, now i pass to CopyOnWriteInputFormat only necessary params, not the full flink configuration. Is that what you want? Or you mean not to pass configuration object at all and pass every configuration parameter (UTC_TIMEZONE, HIVE_STYLE_PARTITIONING, PARTITION_DEFAULT_NAME, PARTITION_PATH_FIELD) as a separate argument of constructor and add  to CopyOnWriteInputFormat as a property?

Reply:
can we revert the change directly, pass around the flat parmas as it is.

Reply:
ok, reverted changes and add two additional flat params, also fixed change log description respectively. just like that?